http-plugins/src/main/java/co/cask/hydrator/plugin/batch/HttpCallbackAction.java

```diff
@@ -72,31 +72,28 @@ public void run(BatchActionContext batchActionContext) throws Exception {
     Exception exception = null;
     do {
       HttpURLConnection conn = null;
-      Map<String, List<String>> headers = conf.parseRequestProperties();
+      Map<String, String> headers = conf.getRequestHeadersMap();
       try {
-        URL url = new URL(conf.url);
+        URL url = new URL(conf.getUrl());
         conn = (HttpURLConnection) url.openConnection();
-        conn.setRequestMethod(conf.method);
-        conn.setConnectTimeout(conf.connectTimeoutMillis);
-        for (Map.Entry<String, List<String>> propertyEntry : headers.entrySet()) {
-          String propertyKey = propertyEntry.getKey();
-          for (String propertyValue : propertyEntry.getValue()) {
-            conn.addRequestProperty(propertyKey, propertyValue);
-          }
+        conn.setRequestMethod(conf.method.toUpperCase());
+        conn.setConnectTimeout(conf.getConnectTimeout());
+        for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+          conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
         }
         if (conf.body != null) {
           conn.setDoOutput(true);
           try (OutputStream outputStream = conn.getOutputStream()) {
             outputStream.write(conf.body.getBytes(Charsets.UTF_8));
```

Comment:
its just a property of json, since all the plugin configs are part of the app config in the create app request (http://docs.cdap.io/cdap/current/en/reference-manual/http-restful-api/lifecycle.html#create-an-application)